### ISSUE TITLE
Add dataset does not work for Exasol backend when using pydobc based sqlalchemy dialect

### ISSUE THREAD

**Issue by Nicoretti** [apache/superset#20105]
Add dataset does not work for Exasol DB when using pydobc based sqlalchemy dialect. 
(see also https://github.com/exasol/sqlalchemy-exasol/issues/136)

#### How to reproduce the bug

0. Setup Exasol DB and add it as data source

1. Go to 'Datasets'
2. Click on '+ DATASET'
3. Select the Exasol DB as database
4. Select a schema containing views and tables
5. Try to select a Table/Schema
6. No Tables/Schemas are shown

### Expected results
Table(s) and view(s) of the specified schema are shown.

### Actual results
No table(s) nor view(s) of the specified schema are shown.

#### Screenshots
**Expected result:**
![image](https://user-images.githubusercontent.com/600911/168984838-bfdafc40-94bf-4b50-957f-d8a5ba327ae0.png)

**Actual result**
![image](https://user-images.githubusercontent.com/600911/168985019-2aec77b8-6f6a-49f7-97dd-060b8e794c56.png)

### Log
```
2022-05-18 09:49:43,443:INFO:werkzeug: * Running on http://127.0.0.1:8000/ (Press CTRL+C to quit)
2022-05-18 09:49:46,313:INFO:werkzeug:127.0.0.1 - - [18/May/2022 09:49:46] "GET /api/v1/me/ HTTP/1.1" 200 -
2022-05-18 09:49:48,206:INFO:werkzeug:127.0.0.1 - - [18/May/2022 09:49:48] "GET /api/v1/database/?q=(filters:!((col:database_name,opr:ct,value:%27%27)),order_columns:database_name,order_direction:asc,page:0,page_size:100) HTTP/1.1" 200 -
2022-05-18 09:49:49,715:INFO:werkzeug:127.0.0.1 - - [18/May/2022 09:49:49] "GET /api/v1/database/?q=(filters:!((col:database_name,opr:ct,value:%27%27)),order_columns:database_name,order_direction:asc,page:0,page_size:100) HTTP/1.1" 200 -
2022-05-18 09:49:49,805:INFO:werkzeug:127.0.0.1 - - [18/May/2022 09:49:49] "GET /api/v1/database/1/schemas/?q=(force:!f) HTTP/1.1" 200 -
Attempt to use a closed connection.
2022-05-18 09:49:52,059:WARNING:superset.models.core:Attempt to use a closed connection.
Attempt to use a closed connection.
2022-05-18 09:49:52,112:WARNING:superset.models.core:Attempt to use a closed connection.
2022-05-18 09:49:52,125:INFO:werkzeug:127.0.0.1 - - [18/May/2022 09:49:52] "GET /superset/tables/1/test_get_metadata_functions_schema/undefined/false/ HTTP/1.1" 200 -
```

### Environment

- browser type and version: `Firefox 100.0`
- superset version: `superset version` = `1.5.0`
- python version: `python --version` = `3.9.7`
- node.js version: `node -v` == `-`
- any feature flags active: `exasol`

### Checklist
- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

This issue was discovered within the context of another issue (#16541). The reporter of that issue mentions that only the version`sqlalchemy-exasol 2.0.10` is working properly together with `superset`.

We already have investigated this issue and can confirm that the cause of this issue is a bug in the exasol sqlalchemy dialect. For further details on the bug in the exasol dialect see also [sqlalchemy-exasol#issue-136](https://github.com/exasol/sqlalchemy-exasol/issues/136).

A fix for the `sqlalchemy-exasol` issue is on the way, see [this PR](https://github.com/exasol/sqlalchemy-exasol/pull/137).

Once the PR is merged and a new version of [sqlalchemy-exasol](https://pypi.org/project/sqlalchemy-exasol/) have been released, adjusting the version constrains of `sqlalchemy-exasol` within `superset` will fix this bug.
